### DISCUSSION TITLE
Metal Slug 6 Debug Menu problem

### DISCUSSION THREAD

**Discussion by Rad1anT1327** [flyinghead/flycast#1146] [Q&A]
Can I make the settings made in the Debug Menu save across restarts?

**Comment by flyinghead** [comment]
Make sure to exit the test menu after making a change.
Also make sure Flycast can write to its `data` folder where flash memory and eeproms are saved.

**Reply by Rad1anT1327** [reply]
It is the Debug Menu of the game, not the Test Menu.  I want to remove blood censorship in Metal Slug 6, but every time I restart, all the settings from the Debug Menu are reset, and the combination to activate is not easy.

**Reply by cinnamonmatexfce** [reply]
in-game menu should keep your preferences until eeprom is not deleted, invalidated.
You could also workaround it by loading a Save State when you boot up the rom

**Reply by Rad1anT1327** [reply]
I could live with this, but I'm confused by the fact that you can download nvmem files from the Internet, in which this censorship of blood, which is only available in the Debug Menu, is removed, and it does not reset.  How did they do it?

**Reply by flyinghead** [reply]
Flycast does save the test menu settings in the corresponding .nvmem/.nvmem2 files. I tested it earlier today before replying.

It looks like it's not doing it on your system for some reason. Can Flycast write to its `data` folder? Do the flycast settings save correctly? Can you save a state?

**Reply by Rad1anT1327** [reply]
Everything works correctly, but you did not understand that the Debug Menu of the game and the Test Menu are different things.

**Reply by flyinghead** [reply]
Right. The debug menu doesn't save anything to the flash memory so the setting is only changed for the current session.
